optaplanner-docs/src/main/asciidoc/Chapter-Optimization_algorithms/Chapter-Optimization_algorithms.adoc

```diff
@@ -604,31 +599,30 @@ Use the `isFeasible()` method on ``BestSolutionChangedEvent``'s new best `Score`
     });
 ----
 
-Use `Score.isSolutionInitialized()` instead of `Score.isFeasible()` to only ignore uninitialized solutions, but do accept infeasible solutions too.
+Use `Score.isSolutionInitialized()` instead of `Score.isFeasible()` to only ignore uninitialized solutions, but also accept infeasible solutions.
 
 [WARNING]
 ====
-The `bestSolutionChanged()` method is called in the solver's thread, as part of ``Solver.solve()``.
+The `bestSolutionChanged()` method is called in the solver's thread, as part of `Solver.solve()`.
 So it should return quickly to avoid slowing down the solving.
 ====
 
 
 [[customSolverPhase]]
 == Custom Solver Phase
 
-Between phases or before the first phase, you might want to run a custom optmization algorithm to initialize the `Solution`
-or to take some low hanging fruit to get a better score quickly.
-Yet you'll still want to reuse the score calculation.
-For example, to implement a custom Construction Heuristic without implementing an entire ``Phase``.
+Run a custom optmization algorithm etween phases or before the first phase to initialize the `Solution`, or to get a better score quickly.
+You will still want to reuse the score calculation.
+For example, to implement a custom Construction Heuristic without implementing an entire `Phase`.
 
 [NOTE]
 ====
-Most of the time, a custom solver phase is not worth the hassle.
+Most of the time, a custom solver phase is inefficient.
```

Comment:
What you have said there is a much better phrase. "Not worth the development time investment" is much better than "not worth the hassle", which is a bit too casual.